pkg/dcgm/health.go

```diff
@@ -23,14 +40,84 @@ type DeviceHealth struct {
 	Watches []SystemWatch
 }
 
-func setHealthWatches(groupId GroupHandle) (err error) {
-	result := C.dcgmHealthSet(handle.handle, groupId.handle, C.DCGM_HEALTH_WATCH_ALL)
+// HealthSet enable the DCGM health check system for the given systems
+func HealthSet(groupId GroupHandle, systems HealthSystem) (err error) {
+	result := C.dcgmHealthSet(handle.handle, groupId.handle, C.dcgmHealthSystems_t(systems))
 	if err = errorString(result); err != nil {
-		return fmt.Errorf("Error setting health watches: %s", err)
+		return fmt.Errorf("Error setting health watches: %w", err)
 	}
 	return
 }
 
+// HealthGet retrieve the current state of the DCGM health check system
+func HealthGet(groupId GroupHandle) (HealthSystem, error) {
+	var systems C.dcgmHealthSystems_t
+
+	result := C.dcgmHealthGet(handle.handle, groupId.handle, (*C.dcgmHealthSystems_t)(unsafe.Pointer(&systems)))
+	if err := errorString(result); err != nil {
+		return HealthSystem(0), err
+	}
+	return HealthSystem(systems), nil
+}
+
+type IncidentError struct {
```

Comment:
I would appreciate if you include a `Error() string` implementation as suggested by GoLang error interface:
```
type error interface {
  Error() string
}

```